cmd/autoheal/alerts_worker.go

```diff
@@ -219,9 +219,10 @@ func (h *Healer) runRule(rule *autoheal.HealingRule, alert *alertmanager.Alert)
 	// Execute the action:
 	switch typed := action.(type) {
 	case *autoheal.AWXJobAction:
-		err = h.runAWXJob(rule, typed, alert)
+		err = h.jobRunners[AWX].RunJob(rule, typed, alert)
+
 	case *batch.Job:
-		err = h.runBatchJob(rule, typed, alert)
+		err = h.jobRunners[BATCH].RunJob(rule, typed, alert)
```

Comment:
What is the benefit of this map if we always index it with constants? Wouldn't it be simpler to have two fields?

```go
type Healer struct {
  ...
  JobRunner awxRunner;
  JobRunner batchRunner;
  ...
}

Reply:
I just thought it would be easier to scale with a map if we eventually in the future add Runners..